src/main/java/com/example/training/redis/FilterRequest.java

```diff
@@ -0,0 +1,92 @@
+package com.example.training.redis;
+
+import com.example.training.security.UserDetailsServiceImpl;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Optional;
+
+@Component
+@Slf4j
+@AllArgsConstructor
+public class FilterRequest extends OncePerRequestFilter {
+    private final UserLimitService userLimitService;
+    public static final int LIMIT_GET = 50;
+    public static final int LIMIT_PUT = 5;
+    private final UserDetailsServiceImpl userDetailsService;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    @NonNull HttpServletResponse response,
+                                    @NonNull FilterChain filterChain) throws ServletException, IOException {
+        String method = request.getMethod();
+        String login = SecurityContextHolder.getContext().getAuthentication().getName();
+        UserDetails userDetails = userDetailsService.loadUserByUsername(login);
+        LocalDate countExpiredDay = getCountExpiredDay(login);
+        if (userDetails != null) {
+            Optional<UserLimit> item = userLimitService.find(userDetails.getUsername());
+            if (item.isEmpty()) {
+                userLimitService.save(userDetails.getUsername(), 1, 1);
+            } else {
+                updateCount(login, method, request, response, filterChain);
+                assert countExpiredDay != null;
+                if (countExpiredDay.equals(LocalDate.now())) {
```

Comment:
why do you update counter first then check the expiration date? If counter date is expired you have to set counter 0 for him